clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -461,10 +461,15 @@ private int append(Map<TopicPartition, List<ConsumerRecord<K, V>>> drained,
     private RequestFuture<Long> sendListOffsetRequest(final TopicPartition topicPartition, long timestamp) {
         Map<TopicPartition, ListOffsetRequest.PartitionData> partitions = new HashMap<>(1);
         partitions.put(topicPartition, new ListOffsetRequest.PartitionData(timestamp, 1));
-        PartitionInfo info = metadata.fetch().partition(topicPartition);
+        Cluster cluster = metadata.fetch();
+        PartitionInfo info = cluster.partition(topicPartition);
         if (info == null) {
+            if (!cluster.topics().contains(topicPartition.topic()))
+                log.warn("No metadata available for topic {}, wait for metadata refresh", topicPartition.topic());
```

Comment:
Good point. We could even leave this at DEBUG level if it's not as exceptional as a missing partition.